gateway-service/src/main/java/org/zowe/apiml/gateway/config/WebSecurity.java

```diff
@@ -304,13 +308,17 @@ private List<ClientRegistration> getClientRegistrations() {
     }
 
     public ServerHttpSecurity defaultSecurityConfig(ServerHttpSecurity http) {
+        var gatewayExceptionHandler = applicationContext.getBean(GatewayExceptionHandler.class);
         return http
             .headers(customizer -> customizer.frameOptions(ServerHttpSecurity.HeaderSpec.FrameOptionsSpec::disable))
             .x509(x509 -> x509
                 .principalExtractor(X509Util.x509PrincipalExtractor())
                 .authenticationManager(X509Util.x509ReactiveAuthenticationManager())
             )
-            .csrf(ServerHttpSecurity.CsrfSpec::disable);
+            .csrf(ServerHttpSecurity.CsrfSpec::disable)
+            .exceptionHandling(exceptionHandlingSpec -> exceptionHandlingSpec.authenticationEntryPoint(
+                (exchange, ex) -> gatewayExceptionHandler.handleAuthenticationException(exchange, ex))
+            );
```

Comment:
It's not from this PR but there's a troubling TODO comment down on line 330, could that be removed?

Reply:
it seems that security works, I will remove the comment